sale_layout_category_hide_detail/static/src/js/boolean_fa_icon_widget.js

```diff
@@ -0,0 +1,78 @@
+/* Copyright 2019 Tecnativa - Ernesto Tejeda
+ * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+ */
+odoo.define('sale_layout_category_hide_detail.boolean_fa_icon_widget', function (require) {
+    "use strict";
+
+    var core = require('web.core');
+    var AbstractField = require('web.AbstractField');
+    var registry = require('web.field_registry');
+
+    var _t = core._t;
+
+    var BooleanFaIconWidget = AbstractField.extend({
+        className: 'o_boolean_fa_icon_widget',
+        events: {
+            'click': '_toggleValue'
+        },
+        supportedFieldTypes: ['boolean'],
+
+        //--------------------------------------------------------------------------
+        // Public
+        //--------------------------------------------------------------------------
+
+        /**
+         * A boolean field is always set since false is a valid value.
+         *
+         * @override
+         */
+        isSet: function () {
+            return true;
+        },
+
+        //--------------------------------------------------------------------------
+        // Private
+        //--------------------------------------------------------------------------
+
+        /**
+         * Render font-awesome icon based on state
+         *
+         * @override
+         * @private
+         */
+        _render: function () {
+            //set icon class
+            var fa_icons = this.attrs.options.fa_icons;
+            var icon_true = fa_icons && fa_icons.icon_true || 'fa-check-square-o';
+            var icon_false = fa_icons && fa_icons.icon_false || 'fa-square-o';
+            var fa_class = this.value ? icon_true : icon_false;
+            //set tip message
+            var terminology = this.attrs.options.terminology;
+            var hover_true = terminology && _(terminology.hover_true) || _('Click to uncheck');
```

Comment:
In JS, `_()` is the [underscore.js](http://underscorejs.org/) library, not the translation function. To translate, it's `_t()`.

OTOH, [the only legal content of a `_t()` is always a raw string](https://www.odoo.com/documentation/12.0/reference/translations.html#explicit-exports). You cannot use variables within, because the contents are processed statically to produce the `.pot` file.

But you cannot translate from the `options` attribute when declaring the widget in the XML view file. How to work it around?

I imagine you got confused because I told you to look at some code that turned out to be buggy, as you can see in https://github.com/odoo/odoo/pull/34774.

What can you do here? `this.nodeOptions.terminology` should be a key from a frozen set of choices, like "check" and "show". [Then, the code, similar to what the `FieldBooleanButton` widget does here](https://github.com/odoo/odoo/blob/04d9715766adab97dd6da66c97e50fd6a30dab31/addons/web/static/src/js/fields/basic_fields.js#L2112-L2124).

Reply:
@Yajo , check what i did and tell me what you think. I have done a separate 'Commit' with some changes to be able to translate the tooltip messages in order to make it more customizable. I know it works but I want your opinion.

Reply:
It's got a bad detail that I commented below in the line.

Apart from that, event when this works fine and is a good solution, I still think that [doing this](https://github.com/odoo/odoo/blob/04d9715766adab97dd6da66c97e50fd6a30dab31/addons/web/static/src/js/fields/basic_fields.js#L2112-L2124) will be better for our use case. It's just more standard code, and for our case at hand it's enough to get the job done, so I think it's better not add any more complexity than the one needed for now.